raftis/src/jepsen/raftis.clj

```diff
@@ -0,0 +1,115 @@
+(ns jepsen.raftis
+  (:gen-class)
+  (:require [clojure.tools.logging :refer :all]
+            [clojure.string :as str]
+            [taoensso.carmine :as car :refer (wcar)]
+            [knossos.model :as model]
+            [jepsen [checker :as checker]
+                    [cli :as cli]
+                    [client :as client]
+                    [control :as c]
+                    [db :as db]
+                    [generator :as gen]
+                    [nemesis :as nemesis]
+                    [tests :as tests]
+                    [util :as util :refer [timeout]]]
+            [jepsen.checker.timeline :as timeline]
+            [jepsen.control.util :as cu]))
+
+(defn r   [_ _] {:type :invoke, :f :read, :value nil})
+(defn w   [_ _] {:type :invoke, :f :write, :value (rand-int 5)})
+
+(defn parse-long
+  "Parses a string to a Long. Passes through `nil`."
+  [s]
+  (when s (Long/parseLong s)))
+
+(defn client
+  "A client for a single register"
+  [conn]
+  (reify client/Client
+    (setup! [_ test node]
+      (def redis-conn {:pool {} :spec {:host node :port 6379 :timeout-ms 5000}})
+      (client redis-conn))
+
+    (invoke! [this test op]
+      (try
+        (case (:f op)
+          :read (assoc op :type :ok, :value (parse-long (get (car/wcar conn (car/get "r")) 0 "")))
+
+          :write (do (car/wcar conn (car/set "r" (:value op)))
+                     (assoc op :type, :ok)))
+
+        (catch clojure.lang.ExceptionInfo e
+          (def err_str (str (.getMessage e)))
+          (def no_leader (re-find #"ERR write InComplete: no leader node!.*" err_str))
+          (def socket_closed (re-find #"socket closed.*" err_str))
+          (assoc op :type (if (or (or (= :read (:f op)) no_leader) socket_closed) :fail :info), :error err_str))
+
+        (catch java.net.SocketTimeoutException e
+          (assoc op :type (if (= :read (:f op)) :fail :info), :error :timeout))
+
+        (catch java.io.EOFException e
+          (assoc op :type :fail, :error :eof_exception))
+
+        (catch java.lang.NumberFormatException e
+          (assoc op :type :fail, :error (str "readnil--- " e)))))
+
+    (teardown! [_ test])))
+
+(def src_dir "/home/gaodunqiao/raftis")
+(def dir     "/opt/raftis")
+(def logfile (str dir "/data/LOG"))
+(def binary  "raftis")
+
+(defn db
+  "Raftis DB for a particular version."
+  [version]
+  (reify db/DB
+    (setup! [_ test node]
```

Comment:
It looks like you haven't provided a way to install the database; without code to do that, other users won't be able to run your test.